syscall_bench_test.go

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2025 The Ebitengine Authors
+
+package purego
+
+import (
+	"fmt"
+	"runtime"
+	"testing"
+)
+
+// shouldSkipCallback returns true if callback tests should be skipped for the given arg count
+func shouldSkipCallback(nargs int) bool {
+	// Callbacks with >15 args have architectural limitations on all platforms
+	// The callback assembly only saves register arguments, not stack arguments
+	return nargs > 15
+}
+
+// BenchmarkCallingMethods compares RegisterFunc, SyscallN, and Callback methods
+func BenchmarkCallingMethods(b *testing.B) {
+	testCases := []struct {
+		n     int
+		fn    any
+		fnPtr uintptr
+		args  []uintptr
+	}{
+		{1, noop1, 0, []uintptr{1}},
+		{2, noop2, 0, []uintptr{1, 2}},
+		{3, noop3, 0, []uintptr{1, 2, 3}},
+		{5, noop5, 0, []uintptr{1, 2, 3, 4, 5}},
+		{10, noop10, 0, []uintptr{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}},
+		{15, noop15, 0, []uintptr{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15}},
+		{16, noop16, 0, []uintptr{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16}},
+		{17, noop17, 0, []uintptr{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17}},
+		{18, noop18, 0, []uintptr{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18}},
+		{19, noop19, 0, []uintptr{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19}},
+		{20, noop20, 0, []uintptr{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20}},
+	}
+
+	// Create callbacks once for all tests
+	for i := range testCases {
+		testCases[i].fnPtr = NewCallback(testCases[i].fn)
+	}
+
+	// Benchmark RegisterFunc (type-safe reflection-based wrapper)
+	b.Run("RegisterFunc", func(b *testing.B) {
+		for _, tc := range testCases {
+			b.Run(fmt.Sprintf("%dargs", tc.n), func(b *testing.B) {
+				if tc.n > 15 {
+					b.Skipf("RegisterFunc with >15 args requires custom implementation")
+				}
+				if shouldSkipCallback(tc.n) {
```

Comment:
Why is `tc.n > 15` checked twice here?

Reply:
it's a mistake, fixing

Reply:
was experimenting with >15 args but that's not needed in this benchmark, cleaning up.